src/utils/service/__spec__.js

```diff
@@ -174,24 +174,124 @@ describe('Service', () => {
     };
 
     describe('errors', () => {
-      it('calls the GET endpoint', (done) => {
-        service.get(5, successSpy, errorSpy);
-        testEndpoint(done, 400, errorSpy);
+      describe('GET with function params', () => {
+        it('calls the GET endpoint', (done) => {
+          spyOn(Logger, 'deprecate');
+          service.get(5, successSpy, errorSpy);
+          testEndpoint(done, 400, errorSpy);
+          expect(Logger.deprecate).toHaveBeenCalled();
+        });
       });
 
-      it('calls the POST endpoint', (done) => {
-        service.post({}, successSpy, errorSpy);
-        testEndpoint(done, 400, errorSpy);
+      describe('GET with object params', () => {
+        it('calls the GET endpoint with query params', (done) => {
+          service.get(5,
+            {
+              params: { foo: 'bar' },
+              onSuccess: successSpy,
+              onError: errorSpy
+            });
+          testEndpoint(done, 400, errorSpy);
```

Comment:
Does the test still pass if you don't pass any params? i.e. are you actually testing that the params are being used/handled? Just curious

Reply:
The test passes if there are no params passed which ensures that they're being defaulted to `{}`. That's as much as it's being tested as the rest is down to `axios`. There are test cases for passing  `params` in the object and not.